File: mentecart/backend/src/models/cart.model.ts
import mongoose from "mongoose";



const cartItemsSchema = new mongoose.Schema(
    {
        service:{
            type: mongoose.Schema.Types.ObjectId,
            ref:"Service",
            required:true,
        },

        date:{
            type: String,
            required: true,
        },

        timeSlot:{
            type: String,
            required: true,
        },

        quantity:{
            type: Number,
            required: true,
        },
        price:{
            type:Number,
            required:true,
        },
    },

    {
        _id:true,
    }
);

const cartSchema = new mongoose.Schema(
    {
        user:{
            type:mongoose.Schema.Types.ObjectId,
            ref: "User",
            required: true,
            unique:true,
        },

        items:[cartItemsSchema],
    },
    {
        timestamps:true,
    }
);

export const Cart = mongoose.model(
    "Cart",
    cartSchema
);

File: mentecart/mobile/lib/features/cart/data/models/cart_model.dart
class CartItemModel {

  final String id;

  final String serviceTitle;

  final String image;

  final int quantity;

  final int price;

  final String date;

  final String timeSlot;

  CartItemModel({
    required this.id,
    required this.serviceTitle,
    required this.image,
    required this.quantity,
    required this.price,
    required this.date,
    required this.timeSlot,
  });

  factory CartItemModel.fromJson(
    Map<String, dynamic> json,
  ) {

    return CartItemModel(
      id: json['_id'],

      serviceTitle:
          json['service']['title'],

      image:
          json['service']['image'],

      quantity: json['quantity'],

      price:
          json['service']['price'],

      date: json['date'],

      timeSlot: json['timeSlot'],
    );
  }
}